src/Microsoft.IdentityModel.Tokens/Encryption/AuthenticatedEncryptionProvider.cs

```diff
@@ -210,6 +224,29 @@ public virtual byte[] Decrypt(byte[] ciphertext, byte[] authenticatedData, byte[
             }
         }
 
+        /// <summary>
+        /// Calls <see cref="Dispose(bool)"/> and <see cref="GC.SuppressFinalize"/>
+        /// </summary>
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
```

Comment:
> GC.SuppressFinalize(this); [](start = 11, length = 27)

class doesn't have a finalizer, do you need this?

Reply:
@henrik-me I'm not sure, every sample of the dispose pattern shows this.

Reply:
@henrik-me the strongest argument I can find is if a derived class in the future adds a finalizer.

Reply:
ack, then that derived class should implement it though. Anyway, I think the compiler will optimize it anyway so I think it's fine

---
In reply to: [396725305](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/1358#discussion_r396725305) [](ancestors = 396725305)